src/test/java/programminglife/parser/GraphParserTest.java

```diff
@@ -55,29 +51,46 @@ public void tearDown() throws Exception {
         faultyGraphParser.getGraph().removeCache();
     }
 
-    @AfterClass
-    public static void tearDownClass() throws Exception {
-        Cache.removeDB(TEST_DB);
-        Cache.removeDB(TEST_FAULTY_DB);
-        Cache.removeDB(TEST_DB2);
-        Cache.removeDB(TEST_FAULTY_DB2);
-    }
-
     @Test(expected = UnknownTypeException.class)
     public void faultyParseTest() throws Exception {
         faultyGraphParser.parse();
     }
 
     @Test
     public void parseTest() throws Exception {
+        GenomeGraph g = graphParser.getGraph();
+
         graphParser.parse();
-        GenomeGraph graph = graphParser.getGraph();
+        for (int i = 1; i < 9; i++)
+            assertTrue(g.contains(i));
+        assertEquals(8, g.size());
+    }
 
-        assertEquals(8, graph.size());
+    @Test
+    public void parseFromCache() throws Exception {
+        GenomeGraph g1 = graphParser.getGraph();
+
+        graphParser.parse();
+        for (int i = 1; i < 9; i++)
```

Comment:
I think `i <= 8` would be clearer
That would also allow for using a constant that can be changed if someone ever decides to change the number of nodes in the test graph.